elide-core/src/main/java/com/yahoo/elide/core/ErrorObjects.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2018, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Error Objects, see http://jsonapi.org/format/#error-objects. <br><br>
+ * Builder example: <br>
+ * <pre><code>
+ *   ErrorObjects errorObjects = ErrorObjects.create()
+ *       .addError()
+ *           .withDetail("first error message")
+ *       .addError()
+ *           .withDetail("second error message")
+ *       .build();
+ * </code></pre>
+ */
+public class ErrorObjects {
+    @Getter private final List<Map<String, Object>> errors;
+
+    public ErrorObjects(List<Map<String, Object>> errors) {
+        this.errors = Objects.requireNonNull(errors, "errors must not be null");
+    }
+
+    public static ErrorObjectsBuilder create() {
```

Comment:
This should be a static method inside `ErrorObjectsBuilder`, it also feels weird to create and `ErrorObjects` in the builder before we call `build()`.